security/src/main/java/com/facebook/airlift/security/ReloadableSslContext.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.airlift.security;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.airlift.security.pem.PemReader;
+import com.google.common.hash.HashCode;
+import com.google.common.io.Files;
+
+import javax.inject.Inject;
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+
+import static com.facebook.airlift.concurrent.Threads.daemonThreadsNamed;
+import static com.google.common.hash.Hashing.sha256;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.Executors.newSingleThreadScheduledExecutor;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public final class ReloadableSslContext
+        implements Supplier<SSLContext>
+{
+    private static final Logger log = Logger.get(ReloadableSslContext.class);
+
+    private final FileWatch trustCertificatesFileWatch;
+    private final FileWatch clientCertificatesFileWatch;
+
+    private final AtomicReference<SSLContext> sslContext;
+    private final ScheduledExecutorService sslContextRefresh;
+
+    @Inject
```

Comment:
`File trustCertificatesFile`, `File clientCertificatesFile` most likely are not gonna be injectable, as both of them of type of `File`.

Also It doesn't feel like the `ReloadableSslContext` is expected to be a bindable generic singleton. There could be multiple `ReloadableSslContext` instances with different settings (different certificates, refresh requirements).

For now I would simply drop the Guice integration.